### ISSUE TITLE
Inlined method with return type "this.type" is ignored, when called several times

### ISSUE THREAD

**Issue by edolgy** [lampepfl/dotty#13035]
## Note

This is the same issue as [12829](https://github.com/lampepfl/dotty/issues/12906) in 3.0.0 and [12906](https://github.com/lampepfl/dotty/issues/12906) in 3.0.1.RC2, both were closed, marked as fixed. But the test fails in newly released 3.0.1. This is clearly not a feature, but rather a very unexpected and hard to detect bug. It needs to be dealt with ASAP.

## Compiler version

3.0.1

## Minimized code

```Scala
class Foo:
  var buf = ""

  def add(v: String): Unit = buf += v

  @scala.annotation.targetName("addOp") inline def += (inline v: String): this.type = {add(v); this }


def main(sa: Array[String]): Unit =

  var fooVar = new Foo
  fooVar += "a" += "b" += "c"
  println(fooVar.buf)              // Prints: abc

  val fooVal = new Foo
  fooVal += "a" += "b" += "c"
  println(fooVal.buf)              // Prints: c     // Expected abc
```
## Problem
If inlined method "+=" return type changed from "this.type" to Foo, the problem goes away.

**Comment by smarter** [comment]
Sorry if what I said in https://github.com/lampepfl/dotty/issues/12906#issuecomment-866270704 wasn't clear enough: the bug was fixed in master which will soon become 3.0.2-RC1 and then eventually 3.0.2, but it was not backported to the 3.0.1 branch, and now that 3.0.1 is released it's too late for that.

**Comment by smarter** [comment]
You can confirm that it's fixed by using a nightly: `scalaVersion := "3.0.2-RC1-bin-20210708-7627583-NIGHTLY"`
